### PR TITLE
ON-5293 # Added Client CLI

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: docs/cdn/deployment-keys.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+- `BLINKM_ACCESS_KEY`
+- `BLINKM_SECRET_KEY`
+
+### Example
+
+```
+export BLINKM_ACCESS_KEY=123456abcdef
+export BLINKM_ACCESS_KEY=abcdefghijklmnopqrst0123456789abcdefghij
+bm client deploy
```
Comment (mymattcarroll) [inline docs/cdn/deployment-keys.md]:
```suggestion
oneblink cdn deploy
```

File: docs/cdn/overview.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+# OneBlink CDN CLI
+
+This tool is for deploying client side code for web apps to the Blink Mobile CDN.
```
Comment (mymattcarroll) [inline docs/cdn/overview.md]:
```suggestion
This tool is for deploying client side code for web apps to the OneBlink CDN.
```

File: docs/cdn/overview.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+oneblink cdn scope
+```
+
+Will tell you what your bucket name is currently set to.
+
+## Deploying files
+
+### Authentication
+
+Before you're able to deploy to an API Hosting Environment, you will need to be authenticated. This is done by logging in with the OneBlink Identity CLI. Please see: [Identity CLI Usage](https://github.com/blinkmobile/identity-cli#usage) for more information on how to log in.
```
Comment (mymattcarroll) [inline docs/cdn/overview.md]:
```suggestion
Before you're able to deploy to a CDN Hosting Environment, you will need to be authenticated. This is done by logging in with the OneBlink Login Command. Please see: [OneBlink Login](../../login/overview.md) for more information on how to log in.
```

File: docs/cdn/overview.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+Initial settings:
+    scope                 => outputs the current scope
+    scope <S3Bucket>      => sets the bucket
+      --region <S3Region> => optionally sets the region
+      --cwd <path>        => outputs or set the scope in <path>
+      --debug             => output debug information
+
+Deploying client side code:
+
+  The deploy command requires a login to BlinkMobile before use.
```
Comment (mymattcarroll) [inline docs/cdn/overview.md]:
```suggestion
  The deploy command requires a login to OneBlink before use.
```

File: docs/cdn/overview.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  For help on the login and logout commands please see:
+  https://github.com/blinkmobile/identity-cli#usage
```
Comment (mymattcarroll) [inline docs/cdn/overview.md]: Maybe just remove this link

File: lib/commands/cdn/lib/aws-credentials.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+function awsCredentials(
+  cfg /* : Object */,
+  env /* : string */,
+  accessToken /* : string | void */,
+) /* : Promise<Object> */ {
+  const spinner = ora({ spinner: 'dots', text: 'Authenticating...' }).start()
+  return new Promise((resolve, reject) => {
+    request(
+      `${cfg.service.origin}/v1/service-instances/${cfg.scope}/environments/${env}/aws-role`,
```
Comment (mymattcarroll) [inline lib/commands/cdn/lib/aws-credentials.js]: This should be coming from the tenant object.

File: lib/commands/cdn/lib/deploy/index.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
+'use strict'
+/* @flow */
+
+const path = require('path')
+
+const ora = require('ora')
+const upload = require('@blinkmobile/aws-s3').upload
+
+const BlinkMobileIdentity = require('@blinkmobile/bm-identity')
+const blinkMobileIdentity = new BlinkMobileIdentity()
```
Comment (mymattcarroll) [inline lib/commands/cdn/lib/deploy/index.js]: This should be passed in from the commands

File: lib/commands/cdn/lib/provision-environment.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  env /* : string */,
+  accessToken /* : string | void */,
+) /* : Promise<void> */ {
+  const spinner = ora({
+    spinner: 'dots',
+    text: `Provisioning environment "${env}"...`,
+  }).start()
+  return new Promise((resolve, reject) => {
+    request.post(
+      `${cfg.service.origin}/v1/service-instances/${cfg.scope}/environments/${env}`,
```
Comment (mymattcarroll) [inline lib/commands/cdn/lib/provision-environment.js]: This should be coming from the tenant object.

File: lib/commands/cdn/lib/read.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      if (!cfg.service.origin && !!cfg.tenant) {
+        cfg.service.origin = tenant.origin
+      }
+      cfg.service.origin = cfg.service.origin || tenant.origin
+      cfg.region = tenant.region
+      cfg.tenant = tenant.label
+
+      const matches = cfg.service.origin.match(
+        /https:\/\/client-cli-service(-test)?\.blinkm\.io/,
+      )
```
Comment (mymattcarroll) [inline lib/commands/cdn/lib/read.js]: This can be deleted now.

File: lib/commands/cdn/lib/aws-credentials.js (hunk 6 above)
Comment (Zaxist) [reply lib/commands/cdn/lib/aws-credentials.js]: Oh, so we aren't overriding the origin in blinkmrc anymore? just env var?

File: test/cdn/cli.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+/* @flow */
+'use strict'
+
+const npmBinTester = require('npm-bin-ava-tester')
+const test = require('ava')
+
+npmBinTester(test)
```
Comment (mymattcarroll) [inline test/cdn/cli.js]: this test can be deleted. it is tested else where.

**Pull request merged**